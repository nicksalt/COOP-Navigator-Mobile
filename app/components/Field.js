import React from "react";
import { View, Text, StyleSheet } from "react-native";
import PropTypes from "prop-types";

export default class Field extends React.Component {
  render() {
    return (
      <View style={styles.fieldContainer}>
        <Text style={styles.field}>{this.props.field}</Text>
        {typeof this.props.fieldValue === "string" ? (
          <Text style={styles.fieldValue}>{this.props.fieldValue}</Text>
        ) : (
          <View style={styles.fieldValueObject}>{this.props.fieldValue}</View>
        )}
      </View>
    );
  }
}

Field.propTypes = {
  fieldValue: PropTypes.oneOfType([PropTypes.element, PropTypes.string])
};

const styles = StyleSheet.create({
  fieldContainer: {
    padding: 20
  },
  field: {
    color: "gray",
    fontSize: 10
  },
  fieldValue: {
    fontSize: 22,
    fontWeight: "bold"
  },
  fieldValueObject: {}
});
