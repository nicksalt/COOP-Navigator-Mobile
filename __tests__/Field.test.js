import React from "react";
import { Text } from "react-native";
import Field from "../app/components/Field";
import renderer from "react-test-renderer";

jest.mock("ScrollView", () => "ScrollView");

test("Field snapshot renders correctly", () => {
  let tree = renderer
    .create(<Field field={"Name"} fieldValue={"John"} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
  tree = renderer
    .create(<Field field={"Name"} fieldValue={<Text>John</Text>} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
