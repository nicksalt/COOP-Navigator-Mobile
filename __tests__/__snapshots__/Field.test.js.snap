// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field snapshot renders correctly 1`] = `
<View
  style={
    Object {
      "padding": 20,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "gray",
        "fontSize": 10,
      }
    }
  >
    Name
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 22,
        "fontWeight": "bold",
      }
    }
  >
    John
  </Text>
</View>
`;

exports[`Field snapshot renders correctly 2`] = `
<View
  style={
    Object {
      "padding": 20,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "gray",
        "fontSize": 10,
      }
    }
  >
    Name
  </Text>
  <View
    style={Object {}}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      John
    </Text>
  </View>
</View>
`;
